sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -577,6 +579,10 @@ def test_adaboost_negative_weight_error(model, X, y):
     sample_weight = np.ones_like(y)
     sample_weight[-1] = -10
 
-    err_msg = "sample_weight cannot contain negative weight"
+    with config_context(assume_positive_sample_weights=False):
+        err_msg = "sample_weight cannot contain negative weight"
+        with pytest.raises(ValueError, match=err_msg):
+            model.fit(X, y, sample_weight=sample_weight)
+    err_msg = "There are negative values in sample_weight"
```

Comment:
I think we should probably unify these messages and have the same error message.